tests/integration/test_workflow_db_mgr.py

```diff
@@ -127,3 +132,54 @@ async def test_workflow_param_rapid_toggle(
 
         w_params = dict(schd.workflow_db_mgr.pri_dao.select_workflow_params())
         assert w_params['is_paused'] == '0'
+
+
+async def test_record_only_non_clock_triggers(
+    flow, run, scheduler, complete, db_select
+):
+    """Database does not record wall_clock xtriggers.
+
+    https://github.com/cylc/cylc-flow/issues/5911
+
+    Includes:
+        - Not in DB: A normal wall clock xtrigger (wall_clock).
+        - In DB: An xrandom mis-labelled as wall_clock trigger  DB).
+        - Not in DB: An execution retry xtrigger.
+
+    @TODO: Refactor to use simulation mode to speedup after Simulation
+    mode upgrade bugfixes: This should speed this test up considerably.
+    """
+    rawpoint = '1348'
+    id_ = flow({
+        "scheduler": {
+            'cycle point format': '%Y',
+            'allow implicit tasks': True
+        },
+        "scheduling": {
+            "initial cycle point": rawpoint,
+            "xtriggers": {
+                "another": "xrandom(100)",
+                "wall_clock": "xrandom(100, _=Not a real wall clock trigger)",
+                "real_wall_clock": "wall_clock()"
+            },
+            "graph": {
+                "R1": """
+                    @another & @wall_clock & @real_wall_clock => foo
+                    @real_wall_clock => bar
+                """
+            }
+        },
+    })
+    schd = scheduler(id_, paused_start=False, run_mode='simulation')
+
+    async with run(schd):
+        foo = schd.pool.get_task(ISO8601Point(rawpoint), 'foo')
+
+        # Wait until all xtriggers for foo are satisfied:
+        while not all(foo.state.xtriggers.values()):
+            await asyncio.sleep(1)
```

Comment:
Should wrap this in `async_timeout` to avoid this test hanging indefinitely if something goes wrong.

Try `git grep async_tmeout` for example usage.